src/boards.c

```diff
@@ -115,6 +115,15 @@ void board_teardown(void)
 
   // Stop LF clock
   NRF_CLOCK->TASKS_LFCLKSTOP = 1UL;
+
+  // make sure all pins are back in reset state
+  for (int i = 0; i < 32; ++i)
```

Comment:
great catch, though I think we should use `nrf_gpio_cfg_default()`, which is designed to do so. 

https://github.com/NordicSemiconductor/nrfx/blob/master/hal/nrf_gpio.h#L222